src/EditorFeatures/CSharpTest/RawStringLiteral/RawStringLiteralCommandHandlerTests.cs

```diff
@@ -319,6 +319,146 @@ public void TestReturnInEmptyFile()
+            var v = """before text$$following text""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = """
+                before text
+                $$following text
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnAfterTextInRawStringFollowingText_Interpolated()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $"""before text$$following text {0}""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = $"""
+                before text
+                $$following text {0}
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnOnInterpolationOpenBraceInRawString()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $"""before text$${0} following text""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = $"""
+                before text
+                $${0} following text
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnInsideInterpolationInRawString()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $"""before text{0$$} following text""";
+            """");
+
+        testState.SendReturn(handled: false);
+    }
+
+    [WpfFact]
+    public void TestReturnWithinOpenBracesInterpolationInRawString()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $$$"""before text{[||]{{0}}} following text""";
+            """");
+
+        testState.SendReturn(handled: false);
+    }
+
+    [WpfFact]
+    public void TestReturnWithinCloseBracesInterpolationInRawString()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = $$$"""before text{{{0}}[||]} following text""";
+            """");
+
+        testState.SendReturn(handled: false);
+    }
+
+    [WpfFact]
+    public void TestReturnBeforeEndQuotesInRawString()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = """before text$$""";
+            """");
+
+        testState.SendReturn(handled: true);
+        testState.AssertCodeIs(
+            """"
+            var v = """
+                before text
+                $$
+                """;
+            """");
+    }
+
+    [WpfFact]
+    public void TestReturnWithinEndQuotesInRawString()
+    {
+        using var testState = RawStringLiteralTestState.CreateTestState(""""
+            var v = """before text""$$";
+            """");
+
+        testState.SendReturn(handled: false);
+    }
```

Comment:
needs tests with multi-line literals since hte code seems to be written to handle it.